src/interceptors/XMLHttpRequest/XMLHttpRequestOverride.ts

```diff
@@ -123,6 +124,7 @@ export const createXMLHttpRequestOverride = (
     ) => any = null as any
 
     constructor() {
+      this.id = uuidv4()
```

Comment:
I've moved the request ID to the request itself (out from the `send()` method) so it's accessible within this class.